### ISSUE TITLE
Issues with Retrieval & Reader approach

### ISSUE THREAD

**Issue by Gchand0249** [UKPLab/sentence-transformers#876]
Hi,

We tried to use Retrieval & Reader approach on our domain specific data but we are facing some issues here.

**Approach**

1. We finetuned Bert with MLM objective with 175K lines of unsupervised data for 4-epochs and using MLM BERT as base model trained BI-encoder & cross-encoder from scratch.
2. For BI-encoder & Cross-encoder training we tagged 25K data manually as 'good','bad','ok'.
3. Then using augmentation technique (synonym replacement in sentence-1 and sentence-2) made it as 200k data points. Then trained BI-encoder with SoftMax loss and Cross-encoder with default loss for 4 epochs.

**We added around 7k domain specific vocab to MLM model and around 1k new domain specific vocab to  BI-encoder & cross-encoder using add_tokens() before training it**

**usage**

1. Retrieving top 20 queries from 1500 queries using BI-encoder and filtering them with Cross-encoder by considering only Good and ok
2. Based on probability Score by SoftMax we are sorting result for good and ok.

**Issues**

**- For some cases model unable to exact same query at top. Model ranking it at position 2 and 3.**

**example:**

input query : 'what is SoftMax loss'

output queries:  

1. '********************'
2.  'what is SoftMax loss'

**-  output probability scores from Cross-encoder model are very high ( 0.999234) and for irrelevant result also
Probability score is very high.**

Hoping to get any advice from you to overcome this issues.

Thanks in advance!!!

**Comment by nreimers** [comment]
It depends on your training data. It does not perceive 'what is SoftMax loss' as a good (relevant) answer to: 'what is SoftMax loss'

Sadly cannot provide real advice here
